### DISCUSSION TITLE
V3 deploy, is it possible yet / are there any docs?

### DISCUSSION THREAD

**Discussion by Cstrachan4** [sanity-io/sanity#3350] [Sanity Studio Support] [answered]
Is there any documentation on deployment for v3 as yet? Running the usual `sanity deploy` seems to still look for the v2 `sanity.json` file rather than the `sanity.cli.ts` or `sanity.config.ts` in v3? 

Thanks!

`Error: "~/sanity.json" does not contain a project identifier ("api.projectId"), which is required for the Sanity CLI to communicate with the Sanity API`

**Comment by jkananen** [comment, marked as answer]
It seems you can do `npx @sanity/cli@dev-preview deploy`, but I've got no further details than trying the command once to see it run.

You can start exploring safely with `npx @sanity/cli@dev-preview --version` and take it from there.

**Reply by kmelve** [reply]
Thanks @jkananen! This is the way to do it for now! We'll bring docs and fixes for the CLI soon!

**Reply by Cstrachan4** [reply]
Ahh perfect, thanks

**Comment by AmeyGohil** [comment]
I added following npm scripts in `package.json` to run sanity commands as running them directly on cli was not working
```
"deploy": "sanity deploy"
"deploy-gql": "sanity graphql deploy"
```
So running `npm run deploy` would do it

**Comment by alex289** [comment]
Hey, I recently tried the solutions but I still get this error. Any ideas?

**Reply by alex289** [reply]
I still get this error on @sanity/cli 3.0.0-dev-preview.10. Any updates on this?

**Reply by rexxars** [reply]
This is usually caused by a global installation of `@sanity/cli`. Could you try running `npx sanity deploy` within the studio folder?

**Reply by alex289** [reply]
@rexxars yes I got it now. Thank you 😄

**Reply by XahidEx** [reply]
Still facing 😪
Sanity v3

**Reply by aharby** [reply]
@XahidEx if you still facing the issue check that you have sanity.cli.js (.ts) file. check https://beta.sanity.io/docs/platform/studio/v2-to-v3/cli

**Reply by surjithctly** [reply]
My issue was because I was using import alias `@/`in sanity, but its not supported yet: See: https://github.com/sanity-io/sanity/issues/4489

I also had to change the compiler option from `es5` to `es6`

@rexxars : any plans to support import alias using `tsconfig`? Since next.js uses that, it would be great if sanity supported that out of the box instead of custom vite config.

**Comment by surjithctly** [comment]
I had the same issue because I was using import alias `@/` in sanity, but it's not supported yet: See: https://github.com/sanity-io/sanity/issues/4489

I removed it to normal `./` and I also had to change the compiler option from `es5` to `es6` to make it work.

**Comment by toiluj23** [comment]
If someone have still this issue. i've added to sanity.cli.js the following code:

inside **export default defineCliConfig({...})** I've added the api object.
``
export default defineCliConfig({
name: 'default',
title: 'Your cms name',
  api: {
    projectId: 'xxxx',
  }
})
``

**Comment by taitsengstock** [comment]
I'm facing this issue still unfortunately. 

Here's my sanity.cli.js:

```
/* eslint-disable no-process-env */
import { projectId, dataset } from './sanity.config.js'
import {defineCliConfig} from 'sanity/cli'

console.log('projectId', projectId)
export default defineCliConfig({api: {projectId, dataset}})
```

And error: Error: `sanity.cli.js does not contain a project identifier ("api.projectId"), which is required for the Sanity CLI to communicate with the Sanity API`

I've tried all solutions listed above. Hopefully someone has some ideas!

Thanks :)

**Reply by rexxars** [reply]
The problem may be cause by `sanity.config.js` containing code that is not runnable in Node.js environment - does it work if you define `projectId` and `dataset` in the `sanity.cli.js` statically?
